remoting/WFLY-19992-Enable-configuration-of-selected-XNIO-Remoting-parameters-for-remoting-objects-using-in-both-inbound-and-outbound-communication.adoc

```diff
@@ -0,0 +1,128 @@
+---
+categories:
+- core
+- remoting
+stability-level:
+default
+issue:
+https://github.com/wildfly/wildfly-proposals/issues/667
+feature-team:
+ developer:
+ tadamski
+ sme:
+  -
+ outside-perspective:
+  -
+promotes:
+promoted-by:
+---
+= <INSERT TITLE HERE>
+:author:            Tomasz Adamski
+:email:             tadamski@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+During analysis of various support cases we found out that there is no possibility to configure XNIO/Remoting options for connections that use remoting protocol (f.e. EJB). It it worth emphasizing that this issue doesn't deal merely with the lack of options parsing that has to be implemented. There are many objects that are created by remoting f.e. Endpoints or Channels, and, what is important in the context of this issue, they can be used in both inbound and outbound scenario. So the goal of this issue is to design and document the elegant way of configuring remoting options.
+
+There already is a discussion in {https://issues.redhat.com/browse/WFCORE-6873}[jira] regarding challenges and possible solutions of this problem. In this proposal I will be describing simplest possible fix that was discussed there , that is endpoint shared by inbound and outbound connections.
+
+=== User Stories
+
+User deploys many number of applications on WildFly - the number of them exceeds the default maximum number of channels for remoting protocol. All those application utilize a remoting connection. Users are getting: "Too many channels open" exception. They would like to configure their server so that the application works correctly.
+
+== Issue Metadata
+
+=== Related Issues
+
+https://issues.redhat.com/browse/WFCORE-6873
+
+=== Affected Projects or Components
+https://github.com/wildfly/wildfly-core Remoting component
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+ 
+* users would be able to configure selected options
+* users would be able to rely on one configuration file "standalone*.xml" for in server deployments
+* configuration available in standalone*.xml configuration - no additional configuration files are neccessary
+
+=== Non-Requirements
+
+* configuration of all available xnio/remoting options
```

Comment:
If this is the case, which subset of XNIO/Remoting options are targeted for configuration?